src/main/java/seedu/address/model/person/PersonContainsTagsPredicate.java

```diff
@@ -0,0 +1,46 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Tags} matches any of the keywords given.
+ */
+public class PersonContainsTagsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public PersonContainsTagsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getTags().stream().map(Object::toString)
+                        .collect(Collectors.joining(" ")), "[" + keyword + "]"));
+    }
```

Comment:
```suggestion
    public boolean test(Person person) {
        String personTags = person.getTags()
                .stream()
                .map(Object::toString)
                .collect(Collectors.joining(" "));
        return keywords.stream()
                .anyMatch(keyword ->  StringUtil.containsWordIgnoreCase(personTags, "[" + keyword + "]"));
    }
```

It would be more efficient and clearly if you refactor this part. Perhaps you can add an example explaining the implementation as well, including the concerns that "[" and "]" are necessary for comparison.